overlay/usr/lib/inithooks/bin/wordpress.py

```diff
@@ -4,65 +4,128 @@
 
+# default interactive value is to set no domain/url in wp-config.php
+DEFAULT_DOMAIN = 'NONE'
 
-def usage(s=None):
+
+def usage(s: Optional[str] | getopt.GetoptError = None) -> NoReturn:
+    std_output = sys.stdout
+    exit_code = 0
     if s:
-        print("Error:", s, file=sys.stderr)
-    print("Syntax: %s [options]" % sys.argv[0], file=sys.stderr)
-    print(__doc__, file=sys.stderr)
-    sys.exit(1)
+        exit_code = 1
+        std_output = sys.stderr
+        print("Error:", s, file=std_output)
+    print(f"Syntax: {sys.argv[0]} [options]", file=sys.stderr)
+    print(__doc__, file=std_output)
+    sys.exit(exit_code)
+
+
+def process_domain(domain: Optional[str]) -> str:
+    if domain is None:
+        return ''
+    domain = domain.strip()
+    if domain == 'NONE':
+        return domain
+    if domain == ("http://"):
+        domain = domain[7:]
+    elif domain == ("https://"):
+        domain = domain[8:]
+    domain.strip("/")
+    if not domain:
+        return ''
+    if domain.startswith("http://") or domain.startswith("https://"):
+        return domain
+    else:
+        return f"https://{domain}"
+
 
 def main():
+    opts: list[tuple[str, str]] = []
+
     try:
         opts, args = getopt.gnu_getopt(sys.argv[1:], "h",
-                                       ['help', 'pass=', 'email='])
+                                       ['help', 'pass=', 'email=', 'domain='])
     except getopt.GetoptError as e:
         usage(e)
 
     password = ""
     email = ""
+    domain = ""
     for opt, val in opts:
         if opt in ('-h', '--help'):
             usage()
         elif opt == '--pass':
             password = val
         elif opt == '--email':
             email = val
+        elif opt == '--domain':
+            domain = val
 
-    if not password:
+    d = Dialog('TurnKey Linux - Configuration')
+    if 'd' not in locals():
         d = Dialog('TurnKey Linux - First boot configuration')
+
+    if not password:
         password = d.get_password(
             "Wordpress Password",
             "Enter new password for the Wordpress 'admin' account.")
 
     if not email:
-        if 'd' not in locals():
-            d = Dialog('TurnKey Linux - First boot configuration')
-
         email = d.get_email(
             "Wordpress Email",
             "Please enter email address for the Wordpress 'admin' account.",
             "admin@example.com")
 
     inithooks_cache.write('APP_EMAIL', email)
-    
+
+    if domain != 'NONE':
+        domain = process_domain(domain)
+        while True:
+            domain = d.get_input(
+                'WordPress domain (optional)',
+                "Enter domain to serve, if no protocol prefix, will assume"
+                " https. Enter 'NONE' to not set a domain",
+                DEFAULT_DOMAIN)
+            domain = process_domain(domain)
+            if not domain:
+                d.error("Domain must be set - or 'NONE'")
+            else:
+                break
```

Comment:
this will prompt for domain, even if domain is already preseeded

Reply:
Good catch! Thanks mate. Although after spending way too long trying to get this to work, I've decided to just cut my losses and force users to set a domain.